controllers/metal3.io/dataimage_controller.go

```diff
@@ -140,6 +142,18 @@ func (r *DataImageReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 		return ctrl.Result{Requeue: false}, nil
 	}
 
+	// If DataImage exists, add its ownerReference
```

Comment:
I think the best order to check stuff in would be:

1. paused annotation
2. owner reference
3. finalizer
4. detached host

Reply:
Done